package/standalone_file_logger/standalone_file_logger/src/standalone_file_logger.cc

```diff
@@ -172,45 +174,129 @@ static void save_prev_logging_fs_type_value()
+      && logging_fs_type == LOGGING_FILESYSTEM_NTFS) {
+    return FSTT_FAT_TO_NTFS;
+  }
+  if (logging_fs_type_prev.is_set && logging_fs_type_prev.value == LOGGING_FILESYSTEM_F2FS
+      && logging_fs_type == LOGGING_FILESYSTEM_NTFS) {
+    return FSTT_F2FS_TO_NTFS;
+  }
+  if (logging_fs_type_prev.is_set && logging_fs_type_prev.value == LOGGING_FILESYSTEM_NTFS
+      && logging_fs_type == LOGGING_FILESYSTEM_F2FS) {
+    return FSTT_NTFS_TO_F2FS;
+  }
+  if (logging_fs_type_prev.is_set && logging_fs_type_prev.value == LOGGING_FILESYSTEM_NTFS
+      && logging_fs_type == LOGGING_FILESYSTEM_FAT) {
+    return FSTT_NTFS_TO_FAT;
+  }
+  if (logging_fs_type_prev.is_set && logging_fs_type_prev.value == LOGGING_FILESYSTEM_F2FS
+      && logging_fs_type == LOGGING_FILESYSTEM_FAT) {
+    return FSTT_F2FS_TO_FAT;
+  }
+  return FSTT_UNKNOWN;
+}
 
-  save_prev_logging_fs_type_value();
-
+static void issue_migration_warning()
+{
   const int str_count = 6;
   const int str_max = 128;
 
   const char warning_strs[str_count][str_max] =
-    {"Logging file-system: ************************************************************",
-     "Logging file-system: Detected that the logging file-system was changed to F2FS...",
+    {"Logging file-system: *****************************************************************",
+     "Logging file-system: Detected that the logging file-system was changed to F2FS/NTFS...",
      "Logging file-system: ... this will ERASE any removable media attached to system!",
      "Logging file-system: The file-system will be reformatted on the next reboot...",
      "Logging file-system: ...settings must be persisted for this to take effect.",
-     "Logging file-system: ************************************************************"};
+     "Logging file-system: *****************************************************************"};
 
   for (size_t x = str_count; x > 0; --x)
     sbp_log(LOG_WARNING, warning_strs[x - 1]);
 
   for (size_t x = 0; x < str_count; ++x)
     piksi_log(LOG_WARNING, warning_strs[x]);
+}
+
+static int logging_filesystem_notify(void *context)
+{
+  (void)context;
+
+  piksi_log(LOG_DEBUG,
+            "%s: curr=%d, prev=%d (set=%hhu)\n",
+            __FUNCTION__,
+            logging_fs_type,
+            logging_fs_type_prev.value,
+            logging_fs_type_prev.is_set);
+
+  /*
+   * Transitions that should issue a warning (because data will be erased):
+   *
+   *   None -> F2FS
+   *   None -> NTFS
+   *   F2FS -> NTFS
+   *   NTFS -> F2FS
+   *
+   * Transitions that should NOT cause a warning (no data will be erased, card will not be
+   * migrated):
+   *
+   *   None -> FAT
+   *   F2FS -> FAT
+   *   NTFS -> FAT
+   */
+
+  fs_type_transition_t transition_type = classify_fs_transition();
+
+  switch (classify_fs_transition()) {
+  case FSTT_NO_DIFFERENCE:
+    PK_LOG_ANNO(LOG_DEBUG, "detected no difference in settings values");
+    break;
+  case FSTT_F2FS_TO_FAT: /* Fall through */
```

Comment:
Utilize `__attribute__ ((fallthrough));`?

Reply:
Looks like this is a GCC 7 feature 🙁